doc/indexing.rst

```diff
@@ -523,3 +523,24 @@ labels:
 
     array
     array.get_index('x')
+
+Sort
```

Comment:
`indexing.rst` is already our longest doc page, so I would put this section in "Reshaping and reorganizing data" instead (`reshaping.rst`).